server/core/src/irodsReServer.cpp

```diff
@@ -77,6 +77,35 @@ namespace {
         }
     }
 
+    /// Get the next connection to use for the next rule
+    ///
+    /// \param[in] user The name of the user to proxy into
+    ix::client_connection get_new_connection(const std::optional<std::string>& _user){
+        static std::atomic<int> host_index = 0;
+        static auto &delay_rule_executors = irods::get_advanced_setting<std::vector<boost::any>&>(irods::DELAY_RULE_EXECUTORS_KW);
+        rodsEnv env{};
+        _getRodsEnv(env);
```

Comment:
if you use the default constructor for L102, you can move L86,87 into the **true** part of the if-statement